### PR TITLE
chore: add release-please

### PR DESCRIPTION
**Pull request by RogerHYang**
See [manifest-driven](https://github.com/googleapis/release-please/blob/main/docs/manifest-releaser.md) release-please.

Our configuration:
- Each package will a separate release PR and will be released individually.

### CODE REVIEW

File: .github/workflows/release.yml [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+      - name: Install utilities
+        run: python -m pip install hatch check_wheel_contents twine
+      - name: Build distribution
+        run: hatch build
+      - name: Check wheel contents
+        run: check-wheel-contents dist/*.whl
+      - name: Publish to PyPI
+        env:
+          TWINE_USERNAME: '__token__'
+          TWINE_PASSWORD: ${{ secrets.PYPI_API_TOKEN }}
```
Comment (RogerHYang) [inline .github/workflows/release.yml:59]: @mikeldking can you add an api token to secrets?

File: release-please-config.json [hunk 2]
```diff
@@ -0,0 +1,26 @@
+{
+  "bump-minor-pre-major": true,
+  "bump-patch-for-minor-pre-major": true,
+  "prerelease-type": "beta",
+  "prerelease": true,
+  "separate-pull-requests": true,
```
Comment (RogerHYang) [inline release-please-config.json:6]: only one package is released at a time

File: python/instrumentation/openinference-instrumentation-openai/src/openinference/instrumentation/openai/package.py [hunk 3]
```diff
@@ -0,0 +1,2 @@
+_instruments = ("openai >= 1.0.0",)
+_supports_metrics = False
```
Comment (axiomofjoy) [inline python/instrumentation/openinference-instrumentation-openai/src/openinference/instrumentation/openai/package.py:2]: Are the contents of this file OTEL-related?

**Review by axiomofjoy — APPROVED** [review]

File: release-please-config.json [hunk 4]
```diff
@@ -0,0 +1,26 @@
+{
+  "bump-minor-pre-major": true,
+  "bump-patch-for-minor-pre-major": true,
+  "prerelease-type": "beta",
+  "prerelease": true,
```
Comment (RogerHYang) [inline release-please-config.json:5]: can remove these remove later

File: python/instrumentation/openinference-instrumentation-openai/pyproject.toml [hunk 5]
```diff
@@ -0,0 +1,52 @@
...
+    "opentelemetry-instrumentation",
+    "opentelemetry-semantic-conventions",
+    "openinference-semantic-conventions",
+    "wrapt",
+]
+
+[project.optional-dependencies]
+test = [
+    "openai == 1.0.0",
+]
```
[the remark this answers is not in the document]
Comment (RogerHYang) [reply python/instrumentation/openinference-instrumentation-openai/pyproject.toml:37]: It's a optional dependency. See [here](https://github.com/open-telemetry/opentelemetry-python-contrib/blob/14d7779366ec99800f1e9a1f7137c01bc1aff617/instrumentation/opentelemetry-instrumentation-httpx/pyproject.toml#L35-L39) for example.

File: python/instrumentation/openinference-instrumentation-openai/pyproject.toml [hunk 6]
```diff
@@ -0,0 +1,52 @@
...
+[project.urls]
+Homepage = "https://github.com/Arize-ai/openinference/tree/main/python/instrumentation/openinference-instrumentation-openai"
+
+[tool.hatch.version]
+path = "src/openinference/instrumentation/openai/version.py"
+
+[tool.hatch.build.targets.sdist]
+include = [
+    "/src",
+    "/tests",
```
[the remark this answers is not in the document]
Comment (RogerHYang) [reply python/instrumentation/openinference-instrumentation-openai/pyproject.toml:48]: Not sure. I'm just following the convention. See [here](https://github.com/open-telemetry/opentelemetry-python-contrib/blob/14d7779366ec99800f1e9a1f7137c01bc1aff617/instrumentation/opentelemetry-instrumentation-httpx/pyproject.toml#L55-L56) for example.

File: python/instrumentation/openinference-instrumentation-openai/README.rst [hunk 7]
```diff
@@ -0,0 +1,14 @@
+OpenInference OpenAI Instrumentation
```
[the remark this answers is not in the document]
Comment (RogerHYang) [reply python/instrumentation/openinference-instrumentation-openai/README.rst:1]: I copied from otel, which uses rst. See [here](https://github.com/open-telemetry/opentelemetry-python-contrib/blob/14d7779366ec99800f1e9a1f7137c01bc1aff617/instrumentation/opentelemetry-instrumentation-httpx/README.rst#L1) for example

File: python/instrumentation/openinference-instrumentation-openai/src/openinference/instrumentation/openai/package.py (hunk 3 above)
Comment (RogerHYang) [reply python/instrumentation/openinference-instrumentation-openai/src/openinference/instrumentation/openai/package.py:2]: I copied it from otel. See [here](https://github.com/open-telemetry/opentelemetry-python-contrib/blob/14d7779366ec99800f1e9a1f7137c01bc1aff617/instrumentation/opentelemetry-instrumentation-httpx/src/opentelemetry/instrumentation/httpx/package.py#L16) for example.

**Pull request merged**